planner/core/indexmerge_path.go

```diff
@@ -461,36 +496,6 @@ func (ds *DataSource) generateAndPruneIndexMergePath(indexMergeConds []expressio
 	if mvIndexMergePath != nil {
 		ds.possibleAccessPaths = append(ds.possibleAccessPaths, mvIndexMergePath...)
 	}
-
-	// 4. If needed, append a warning if no IndexMerge is generated.
-
-	// If without hints, it means that `enableIndexMerge` is true
-	if len(ds.indexMergeHints) == 0 {
-		return nil
-	}
-	// With hints and without generated IndexMerge paths
-	if regularPathCount == len(ds.possibleAccessPaths) {
-		ds.indexMergeHints = nil
-		ds.ctx.GetSessionVars().StmtCtx.AppendWarning(errors.Errorf("IndexMerge is inapplicable"))
-		return nil
-	}
-
-	// 4. If needPrune is true, prune non-IndexMerge paths.
-
-	// Do not need to consider the regular paths in find_best_task().
-	// So we can use index merge's row count as DataSource's row count.
-	if needPrune {
```

Comment:
Maybe we can remove the argument `needPrune` now.

Reply:
You are right. It was removed.